### PR TITLE
Staging Sites: staging sync UI enchancements

### PR DESCRIPTION
**Pull request by kozer**
Related to https://github.com/Automattic/dotcom-forge/issues/4135

## Proposed Changes

This pr adds various UI enhancements that were left over from the original Sttaging sync prototype.

### Test 1: UI enchancements
1. Create a WoA site.
2. Add a staging site.
3. Navigate to `Hosting` page
4. In your production site ensure that you see the following UI:
2023-10-10T15:02:47,962693120+03:00
5. Navigate to your staging site
6. Ensure that you see the following UI:
2023-10-10T15:02:36,858480603+03:00

### Test 2: Selected options order
1. While on either the production or staging site, change the option to sync your changes to production ( you should have some available checkboxes.)
2. Check some of the checkboxes, and press the Synchronize button.
3. Ensure that the options you have selected, appear in the same order as they appear originally.
4. Ensure that the checkboxes and items appear in the same order regardless of the navigation between "pull" and "push" selections.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx [hunk 1]
```diff
@@ -149,14 +172,19 @@ function StagingSiteProductionCard( { disabled, siteId, translate }: CardProps )
...
-		<Card className="staging-site-card">
-			{
-				// eslint-disable-next-line wpcalypso/jsx-gridicon-size
-				<Gridicon icon="science" size={ 32 } />
-			}
-			<CardHeading id="staging-site">{ __( 'Staging site' ) }</CardHeading>
-			{ cardContent }
-		</Card>
+		<>
+			<SVGContainer>
```
Comment (wojtekn) [inline client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx]: Would it be more correct to make this component a part of the component it decorates? Also, if it's just decoration, would it be better to have it as a CSS background and not an SVG element placed within the DOM?

Besides that, we should adjust the component naming to follow what the component does e.g. `Divider` and `DividerPattern`.

**Review by wojtekn — COMMENTED** [review]
Items ordering change looks like expected. We still need to adjust the spacing in the new box:

Screenshot 2023-10-11 at 14 10 35

And there is some small glitch in the roadblock pattern, like we miss 1px there:

Screenshot 2023-10-11 at 14 13 03

File: client/my-sites/hosting/staging-site-card/card-content/manage-staging-site-card-content.tsx [hunk 2]
```diff
@@ -168,12 +183,7 @@ export const ManageStagingSiteCardContent = ( {
 							/>
 						</SyncActionsContainer>
 					</>
-				) : (
-					<ActionButtons>
-						<ManageStagingSiteButton />
-						<ConfirmationDeleteButton />
-					</ActionButtons>
-				) }
```
Comment (wojtekn) [inline client/my-sites/hosting/staging-site-card/card-content/manage-staging-site-card-content.tsx:176]: I like this cleanup! DRY

File: client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx (hunk 1 above)
Comment (kozer) [reply client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx]: > Would it be more correct to make this component a part of the component it decorates? 

I did that for practical reasons. For example, we may need to reuse that SVG, but without the need for margin etc as this container provides.

> Besides that, we should adjust the component naming to follow what the component does e.g. Divider and DividerPattern.

Agree! Will do that! Thanks!

Comment (wojtekn) [reply client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx]: > I did that for practical reasons. For example, we may need to reuse that SVG, but without the need for margin etc as this container provides.

I think it should be ok to add it to `client/assets/`.

File: client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx [hunk 3]
```diff
@@ -149,14 +173,17 @@ function StagingSiteProductionCard( { disabled, siteId, translate }: CardProps )
 		);
 	}
 	return (
-		<Card className="staging-site-card">
+		<ProductionCard className="staging-site-card">
+			<DividerContainer>
+				<img src={ dividerPattern } alt="staging site production card divider" />
```
Comment (danielbachhuber) [inline client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx]: The alt text needs localization too.

Comment (wojtekn) [reply client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx]: @kozer this image purpose is decoration. What prevents us from using it as a `background-image` CSS property for the `ProductionCard`?

Comment (kozer) [reply client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx]: > this image purpose is decoration. What prevents us from using it as a background-image CSS property for the ProductionCard

Done! I use it as a background image in the original container, as it uses an svg pattern, and if I add it as a background-image at the `ProductionCard`, it fills the whole card's content.

**Review by wojtekn — APPROVED** [review]
Thanks @kozer , looks good.

**Pull request merged**